### PR TITLE
feat(hx-beacon): implement beacon with component tests

### PR DESCRIPTION
**Pull request by 100stacks**
## Description

* Implement `<hx-beacon>` and component tests
* minor docs update for `Toast` component

### screenshot `<hx-beacon>` component tests

Screen Shot 2020-06-17 at 11 23 58 PM

### What are the relevant story cards/tickets? Any additional PRs or other references?

Jira: SURF-1859, SURF-2041

## Before you request a review for this PR:

- [x] For UI changes, did you manually test in recent versions of modern browsers (Chrome, Firefox, and Safari)?
- [x] For UI changes, did you manually test in IE11 and legacy Edge?
- [x] Did you add component tests for any new code?
- [x] Did you run the component unit tests via `yarn test` to ensure all tests passed?
- [x] Did you include a screenshot of the component tests?
- [x] If you changed/added functionality, did you update the demo page and documentation?
- [ ] If needed, did you add or modify the demo test page to test the changed/added functionality?
- [x] Did you assign reviewers?
- [x] In Jira, have you linked to this PR on the ticket(s)?

### CODE REVIEW

**Comment by 100stacks** [comment]
Continued work from @BY00565233 PR #677.

File: docs/components/beacon/index.html [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+                id="chkBeaconInlineStyles"
+                type="checkbox"
+                v-model="isDisabled"
+              />
+              <label for="chkBeaconInlineStyles">
+                <hx-checkbox></hx-checkbox>
+                Apply styles
+              </label>
+            </hx-checkbox-control> -->
+            <hx-text-control>
```
Comment (100stacks) [inline docs/components/beacon/index.html:72]: ### Simplified this demo based on @lalithkarikelli feedback.  Before, the demo provided an option to disable the inline styles.

Screen Shot 2020-06-18 at 10 24 10 AM

**Comment by 100stacks** [comment]
> it looks like the relative positioning causes the containing div to retain the amount of space used by the beacon. This causes the button to jump up when the beacon is removed.
> 
> It might be necessary to use Absolute positioning on the beacon, and instruct implementors to make sure that the containing div has relative positioning set.

@araiford appreciated the great feedback.  Any thoughts on the `Basic Beacon with Inline Styles` demo?  I wanted to to make it interactive, to show how `<hx-beacon>` can be fine-tuned.

File: src/scss/components/beacon/index.scss [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+/// ========== hxBeacon ========== ///
+hx-beacon {
+  background-color: $gray-0;
+  border-radius: 1rem;
+  box-shadow: 0 0 2px 2px $gray-600;
+  display: block;
+  height: 1rem;
+  margin: 0;
+  padding: 0;
+  position: absolute;  // parent container must be "position: relative"
```
Comment (100stacks) [inline src/scss/components/beacon/index.scss:13]: @araiford, updated based on your feedback.

**Comment by araiford** [comment]
:shipit:

**Pull request merged**